Add 'flit develop' command

jhamrick opened the pull request:
This adds a `flit develop` subcommand which acts the same as `flit install --symlink` with the exception that it also installs any development requirements that are specified in the `flit.ini` file (under `develop-requires`).

I'll add tests and update relevant docs if you think this looks ok.

Carreau commented:
+1

Carreau commented:
Note from :beers: , check if/how to use `test-requirements` metadata for that.

takluyver commented:
I'm happy with the basic idea, but I'm still thinking a bit about what the interface should look like.

takluyver commented on flit/inifile.py:
```diff
@@ -93,7 +94,7 @@ def verify_classifiers(classifiers):
 
 
-def read_pkg_ini(path):
+def read_pkg_ini(path, develop=False):
```
I don't think the options you pass in should reach this level - the `inifile` module is to read and check the config file, and that process shouldn't be affected by what you want to use it for.

jhamrick commented:
Ok, updated as per our conversation in person. Got rid of `flit develop` and modified `flit install` so that it also installs any development requirements, if they are given. Additionally, I added `-s` to the `--symlink` flag so that you can do `flit install -s` to get a proper development install rather than having to type the full `flit install --symlink`.

takluyver commented:
Thanks, I think this looks good.

I'd quite like to have a way to control which dependencies get installed before we release again. You're welcome to work on this if you'd like to, or I'll hopefully get to it soon.
